src/ecrn_util.erl

```diff
@@ -7,12 +7,41 @@
 %%%  Utility functions for the erlcron system
 -module(ecrn_util).
 
--export([epoch_seconds/0]).
+-export([epoch_seconds/0, epoch_milliseconds/0]).
+-export([epoch_to_time_string/1, epoch_to_datetime_string/1]).
+-export([universaltime_to_epoch/1, localtime_to_epoch/1]).
 
 %%%===================================================================
 %%% API
 %%%===================================================================
 -spec epoch_seconds() -> erlcron:seconds().
 epoch_seconds() ->
-    {Megasecs, Secs, Microsecs} = os:timestamp(),
-    erlang:trunc((Megasecs * 1000000) + Secs + (Microsecs / 1000000)).
+    erlang:system_time(seconds).
+
+-spec epoch_milliseconds() -> erlcron:milliseconds().
+epoch_milliseconds() ->
+    erlang:system_time(millisecond).
+
+-spec epoch_to_time_string(erlcron:milliseconds()) -> string().
+epoch_to_time_string(Epoch) when is_integer(Epoch) ->
+    DT = erlang:posixtime_to_universaltime(Epoch div 1000),
+    {_, {H,M,S}} = erlang:universaltime_to_localtime(DT),
+    lists:flatten(io_lib:format("~.2.0w:~.2.0w:~.2.0w.~.3.0w", [H,M,S, Epoch rem 1000])).
+
+-spec epoch_to_datetime_string(erlcron:milliseconds()) -> string().
+epoch_to_datetime_string(Epoch) when is_integer(Epoch) ->
+    DT = erlang:posixtime_to_universaltime(Epoch div 1000),
+    {{Y,Mo,D}, {H,M,S}} = erlang:universaltime_to_localtime(DT),
+    lists:flatten(io_lib:format("~w-~.2.0w-~.2.0w ~.2.0w:~.2.0w:~.2.0w.~.3.0w",
```

Comment:
Can this be a private utility function, the `lists:flatten` and the `io_lib:format` calls?

Reply:
Done